File: src/AtCoder/abc286/c.py
from collections import deque

N, A, B = map(int, input().split())

S = input()

def main():
    dq = deque(S)

    ans = 10**20

    for i in range(N):
        score = i * A
        for j in range(0, N//2):
            if dq[j] != dq[N-1-j]:
                score += B
        ans = min(ans, score)
        dq.append(dq.popleft())

    print(ans)

main()

File: src/AtCoder/abc286/c.rs
#![allow(unused_imports)]
use std::collections::*;
use itertools::Itertools;
use proconio::input;
use proconio::marker::*;

fn main() {
    input! {
        n: usize,
        a: usize,
        b: usize,
        s: Chars,
    }
    let mut ans = std::usize::MAX;
    let mut dq = VecDeque::from(s);
    for i in 0..n {
        let mut score = i * a;
        for j in 0..(n/2) {
            if dq[j] != dq[n-1-j] {
                score += b;
            }
        }
        ans = ans.min(score);
        let c = dq.pop_front().unwrap();
        dq.push_back(c)
    }
    println!("{}", ans);
}
